src/networking.c

```diff
@@ -4417,16 +4324,16 @@ size_t getClientMemoryUsage(client *c, size_t *output_buffer_mem_usage) {
      * i.e. unused sds space and internal fragmentation, just the string length. but this is enough to
      * spot problematic clients. */
     mem += c->argv_len_sum + sizeof(robj *) * c->argc;
-    mem += multiStateMemOverhead(c);
+    mem += c->mstate ? multiStateMemOverhead(c) : 0;
 
     /* Add memory overhead of pubsub channels and patterns. Note: this is just the overhead of the robj pointers
      * to the strings themselves because they aren't stored per client. */
-    mem += pubsubMemOverhead(c);
+    mem += c->pubsub_data ? pubsubMemOverhead(c) : 0;
```

Comment:
Change not needed because pubsubMemOverhead has this check and returns 0.

```suggestion
    mem += pubsubMemOverhead(c);
```